lib/cloud/index.js

```diff
@@ -1,139 +1,101 @@
 'use strict';
 
-var express = require('express'),
-  multer = require('multer'),
-  config = require('../config'),
-  os = require('os'),
-  uuid = require('uuid-js'),
-  q = require('q'),
-  fs = require('fs'),
-  base64 = require('base64-stream'),
-  through = require('through2'),
-  imageDir = os.tmpdir() + '/wfm';
+var express = require('express');
+var config = require('../config');
+var uuid = require('uuid-js');
+var fileService = require('./services/fileService');
 
-fs.mkdir(imageDir, '0775', function(err) {
-  if (err && err.code !== 'EEXIST') {
-    console.log(err);
-    throw new Error(err);
-  }
-});
-
-function writeStreamToFile(fileMeta, stream) {
-  var deferred = q.defer();
-  stream.on('end', function() {
-    // console.log('File save complete:', filename);
-    deferred.resolve(fileMeta);
-  });
-  stream.on('error', function(error) {
-    deferred.reject(error);
-  });
-  var filename = imageDir + '/' + fileMeta.uid;
-  stream.pipe(fs.createWriteStream(filename));
-  return deferred.promise;
-}
-
-var parseBase64Stream = function(req) {
-  var passthrough = false;
-  var accumulation = '';
-  var stream = req.pipe(through(function(chunk, enc, callback) {
-    if (!passthrough) {
-      accumulation += chunk;
-      var test = ';base64,';
-      var index = accumulation.indexOf(test);
-      if (index > - 1) {
-        passthrough = true;
-        chunk = accumulation.substr(index + test.length);
-      }
-    }
-    if (passthrough) {
-      this.push(chunk);
-    }
-    callback();
-  }))
-    .pipe(base64.decode());
-  return stream;
-};
-
-
-function initRouter(mediator) {
-  var router = express.Router();
+function initRouter(mediatorService) {
+  var router = express.Router()
+  
   router.route('/all').get(function(req, res) {
-    mediator.publish(config.cloudDataTopicPrefix + config.datasetId + ':list');
-    mediator.once('done:' + config.cloudDataTopicPrefix + config.datasetId + ':list', function(files) {
-      res.json(files);
+    mediatorService.listFilesMetadata(null).then(function(fileList) {
```

Comment:
The error cases should be handled here also otherwise the request will just time out.